evalml/tests/component_tests/test_estimators.py

```diff
@@ -297,3 +299,60 @@ class MockEstimator(Estimator):
     X, y = est._manage_woodwork(X_df, y_series)
     assert isinstance(X, pd.DataFrame)
     assert isinstance(y, pd.Series)
+
+
+@pytest.mark.parametrize("use_numerical_targets_for_classification", [True, False])
+@pytest.mark.parametrize("estimator_class", _all_estimators())
+@pytest.mark.parametrize("use_custom_index", [True, False])
+@pytest.mark.parametrize(
+    "problem_type",
+    [ProblemTypes.BINARY, ProblemTypes.MULTICLASS, ProblemTypes.REGRESSION],
+)
+def test_estimator_fit_predict_and_predict_proba_respect_custom_indices(
+    use_numerical_targets_for_classification,
+    problem_type,
+    use_custom_index,
+    estimator_class,
+    X_y_binary,
+    X_y_multi,
+    X_y_regression,
+):
+    if estimator_class not in get_estimators(problem_type) or (
+        problem_type == ProblemTypes.REGRESSION
+        and use_numerical_targets_for_classification
+    ):
+        return
+    if problem_type == ProblemTypes.BINARY:
+        X, y = X_y_binary
+    elif problem_type == ProblemTypes.MULTICLASS:
+        X, y = X_y_multi
+    elif problem_type == ProblemTypes.REGRESSION:
+        X, y = X_y_regression
+
+    X = pd.DataFrame(X)
+    y = pd.Series(y)
+
+    if is_classification(problem_type) and not use_numerical_targets_for_classification:
+        y = y.map({val: str(val) for val in np.unique(y)})
+
+    if use_custom_index:
+        gen = np.random.default_rng(seed=0)
```

Comment:
I know we do this in similar tests, but it's bold.  I'm always hesitant to see pseudo-randomness added into a testing suite.  I think it opens the door for potential flakiness and I'm not sure about the implications across platforms.  I'm pretty sure this usage is fine, though.

Reply:
Heh, I had taken inspiration from our other tests but agreed! The pseudo-randomness isn't really necessary here so I'll remove it from this instance and our other tests that use this :)